pkg/issuer/acme/http/solver/solver.go

```diff
@@ -57,7 +58,14 @@ func (h *HTTP01Solver) Listen(log logr.Logger) error {
 
 	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		// extract vars from the request
-		host := strings.Split(r.Host, ":")[0]
+		host, _, err := net.SplitHostPort(r.Host)
```

Comment:
@Muhannadimad can you add a test?

Reply:
@inteon  could you help me with which tests to add and how to do it? sorry this is my first time contributing here.